crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -120,13 +155,62 @@ where
                 .map(|block_randomness| {
                     FraudProofVerificationInfoResponse::BlockRandomness(block_randomness)
                 }),
-            FraudProofVerificationInfoRequest::DomainTimestampExtrinsic(doman_id) => self
-                .derive_domain_timestamp_extrinsic(consensus_block_hash, doman_id)
+            FraudProofVerificationInfoRequest::DomainTimestampExtrinsic(domain_id) => self
+                .derive_domain_timestamp_extrinsic(consensus_block_hash, domain_id)
                 .map(|domain_timestamp_extrinsic| {
                     FraudProofVerificationInfoResponse::DomainTimestampExtrinsic(
                         domain_timestamp_extrinsic,
                     )
                 }),
+            FraudProofVerificationInfoRequest::DomainBundleBody {
+                domain_id,
+                bundle_index,
+            } => self
+                .get_domain_bundle_body(consensus_block_hash, domain_id, bundle_index)
+                .map(|domain_bundle_body| {
+                    FraudProofVerificationInfoResponse::DomainBundleBody(domain_bundle_body)
+                }),
         }
     }
+
+    fn derive_bundle_digest(
+        &self,
+        consensus_block_hash: H256,
+        domain_id: DomainId,
+        bundle_body: Vec<OpaqueExtrinsic>,
+    ) -> Option<H256> {
+        let domain_runtime_code = {
+            let consensus_block_hash = consensus_block_hash.into();
+            self.consensus_client
+                .runtime_api()
+                .domain_runtime_code(consensus_block_hash, domain_id)
+                .ok()??
+        };
+
+        let mut extrinsics = Vec::with_capacity(bundle_body.len());
+        for opaque_extrinsic in bundle_body {
+            let ext = <<DomainBlock as BlockT>::Extrinsic>::decode(
+                &mut opaque_extrinsic.encode().as_slice(),
+            )
+            .ok()?;
+            extrinsics.push(ext);
+        }
+
+        let domain_runtime_api_light =
+            RuntimeApiLight::new(self.executor.clone(), domain_runtime_code.into());
+
+        let ext_signers: Vec<_> = SignerExtractor::<DomainBlock>::extract_signer(
+            &domain_runtime_api_light,
+            // `extract_signer` is a stateless runtime api thus it is okay to use
+            // default block hash
+            Default::default(),
+            extrinsics,
+        )
+        .ok()?
+        .into_iter()
+        .map(|(signer, tx)| (signer, BlakeTwo256::hash_of(&tx)))
```

Comment:
Plese use Block::Hashing

Reply:
The `bundle_digest` is derived on the client side (in `block-preprocessor`) using `BlakeTwo256` so I used the same here to keep consistency, or do you want me to update both places?

Reply:
Yeap. I think we should explictly avoid using Bare hash functions instead use the hash from `Block` instead